homeassistant/components/tplink/const.py

```diff
@@ -1,5 +1,74 @@
 """Const for TP-Link."""
 import datetime
 
+from homeassistant.components.sensor import ATTR_STATE_CLASS, STATE_CLASS_MEASUREMENT
+from homeassistant.components.switch import ATTR_CURRENT_POWER_W, ATTR_TODAY_ENERGY_KWH
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_FRIENDLY_NAME,
+    ATTR_UNIT_OF_MEASUREMENT,
+    ATTR_VOLTAGE,
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+)
+
 DOMAIN = "tplink"
+COORDINATORS = "coordinators"
+
 MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(seconds=8)
+MAX_DISCOVERY_RETRIES = 4
+
+ATTR_CONFIG = "config"
+ATTR_TOTAL_ENERGY_KWH = "total_energy_kwh"
+ATTR_CURRENT_A = "current_a"
+
+CONF_MODEL = "model"
+CONF_SW_VERSION = "sw_ver"
+CONF_EMETER_PARAMS = "emeter_params"
+CONF_DIMMER = "dimmer"
+CONF_DISCOVERY = "discovery"
+CONF_LIGHT = "light"
+CONF_STRIP = "strip"
+CONF_SWITCH = "switch"
+CONF_SENSOR = "sensor"
+
+PLATFORMS = [CONF_LIGHT, CONF_SENSOR, CONF_SWITCH]
+
+ENERGY_SENSORS = {
+    ATTR_CURRENT_POWER_W: {
+        ATTR_UNIT_OF_MEASUREMENT: POWER_WATT,
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_FRIENDLY_NAME: "Current Consumption",
```

Comment:
Should these friendly name attributes be keyed to strings so that they can be translated?  You might even be able to reference existing strings in the generic sensor component, eg: `"[%key:component::sensor::device_automation::condition_type::is_power%]"`.

Reply:
We don't currently support translating names of entities. Adding support for that is out of scope for this PR.

Reply:
Ah. Never mind then.